### PR TITLE
[MLv2] [FE] Migrate object detail

### LINKED ISSUE #37628: [MLv2] [FE] Migrate object detail
https://github.com/metabase/metabase/blob/d3a9d9d6013daf4e5b12bc1b9e83996957e0d151/frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetailView.tsx#L157
https://github.com/metabase/metabase/blob/59bbe347d1710e6d78b1327543a3c9395495f64a/frontend/src/metabase/query_builder/actions/object-detail.js#L59

### PR DESCRIPTION
**Pull request by kamilmielnik**
Closes #37628
Depends on #37648

### CODE REVIEW

File: frontend/src/metabase/lib/card.js [hunk 1]
```diff
@@ -14,10 +14,10 @@ export function createCard(name = null) {
 }
 
 // start a new card using the given query type and optional database and table selections
-export function startNewCard(type, databaseId, tableId) {
+export function startNewCard(databaseId, tableId) {
```
Comment (kamilmielnik) [inline frontend/src/metabase/lib/card.js:17]: This function is getting removed in #37629

File: frontend/src/metabase/query_builder/actions/object-detail.js [hunk 2]
```diff
@@ -56,13 +57,31 @@ export const followForeignKey = createThunkAction(
...
       }
 
-      const newCard = startNewCard("query", card.dataset_query.database);
-
-      newCard.dataset_query.query["source-table"] = fk.origin.table.id;
-      newCard.dataset_query.query.filter = getFilterForFK(objectId, fk);
+      const databaseId = card.dataset_query.database;
+      const tableId = fk.origin.table.id;
+      const metadata = getMetadata(getState());
+      const question = Question.create({ databaseId, tableId, metadata });
```
Comment (uladzimirdev) [inline frontend/src/metabase/query_builder/actions/object-detail.js:63]: could you please clarify why `Question.create` is used instead of new Question?

**Review by uladzimirdev — APPROVED** [review]

File: frontend/src/metabase/query_builder/actions/object-detail.js (hunk 2 above)
Comment (kamilmielnik) [reply frontend/src/metabase/query_builder/actions/object-detail.js]: 9c6a7ecc183337644a075ba17b57e94874905ea5

Updated to not pass `tableId` to `Question.create`. For context see [Slack discussion](https://metaboat.slack.com/archives/C0645JP1W81/p1705322543442019).

File: frontend/src/metabase/query_builder/actions/object-detail.js [hunk 3]
```diff
@@ -56,13 +57,34 @@ export const followForeignKey = createThunkAction(
         return false;
       }
 
-      const newCard = startNewCard("query", card.dataset_query.database);
-
-      newCard.dataset_query.query["source-table"] = fk.origin.table.id;
-      newCard.dataset_query.query.filter = getFilterForFK(objectId, fk);
+      const databaseId = card.dataset_query.database;
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/actions/object-detail.js:60]: Could you create a question from `card` and get `databaseId` this way? We shouldn't be parsing `dataset_query` like this

File: frontend/src/metabase/query_builder/actions/object-detail.js [hunk 4]
```diff
@@ -56,13 +57,34 @@ export const followForeignKey = createThunkAction(
...
+      const tableId = fk.origin.table.id;
+      const metadata = getMetadata(getState());
+      const metadataProvider = Lib.metadataProvider(databaseId, metadata);
+      const table = Lib.tableOrCardMetadata(metadataProvider, tableId);
+      const query = Lib.queryFromTableOrCardMetadata(metadataProvider, table);
+      const stageIndex = -1;
+      const column = Lib.fromLegacyColumn(query, stageIndex, fk.origin);
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/actions/object-detail.js:67]: I'm surprised it worked this way, but seems legit. Could you update `fromLegacyColumn` and include `Field` from `metabase-types/api` as well?

**Review by ranquild — APPROVED** [review]
Small comments, approving

File: frontend/src/metabase/query_builder/actions/object-detail.js (hunk 3 above)
Comment (kamilmielnik) [reply frontend/src/metabase/query_builder/actions/object-detail.js]: 47007574d8792fe8ca40ee59ce4102c97f85f985

File: frontend/src/metabase/query_builder/actions/object-detail.js (hunk 4 above)
Comment (kamilmielnik) [reply frontend/src/metabase/query_builder/actions/object-detail.js:67]: 234b538e2d0d356a2956784dd03582f6c7f69317

**Pull request merged**